src/unlimited.cpp

```diff
@@ -1759,18 +1759,52 @@ UniValue getminingcandidate(const UniValue &params, bool fHelp)
     CMiningCandidate candid;
     int64_t coinbaseSize = -1; // If -1 then not used to set coinbase size
 
-    if (fHelp || params.size() > 1)
+    if (fHelp || params.size() > 2)
     {
-        throw runtime_error("getminingcandidate"
-                            "\nReturns Mining-Candidate protocol data.\n"
-                            "\nArguments:\n"
-                            "1. \"coinbasesize\" (int, optional) Get a fixed size coinbase transaction.\n" +
-                            HelpExampleCli("", "") + HelpExampleCli("coinbasesize", "100"));
+        throw runtime_error(
+            "getminingcandidate"
+            "\nReturns Mining-Candidate protocol data.\n"
+            "\nArguments:\n"
+            "1. \"coinbasesize\" (int, optional) Get a fixed size coinbase transaction.\n"
+            "2. \"address\"      (string, optional) The address to send the newly generated bitcoin to.\n"
+            "                                     Default: an address in daemon's wallet.\n" +
+            HelpExampleCli("getminingcandidate", "") +
+            HelpExampleCli("getminingcandidate", "130 bchtest:qq9rw090p2eu9drv6ptztwx4ghpftwfa0gyqvlvx2q") +
+            HelpExampleCli("getminingcandidate", "bchtest:qq9rw090p2eu9drv6ptztwx4ghpftwfa0gyqvlvx2q"));
     }
 
-    if (params.size() == 1)
+    boost::shared_ptr<CReserveScript> coinbaseScript;
+    std::string destStr;
+    bool parsedCoinbaseSize = false;
+
+    // we accept: param1,param2 or just param1 by itself or just param2 by itself
+    if (params.size() >= 1)
+    {
+        try
+        {
+            // see if first param was an int, if so was coinbaseSize
+            coinbaseSize = UniValue(UniValue::VNUM, params[0].getValStr()).get_int64();
+            parsedCoinbaseSize = true;
+        }
+        catch (...)
+        {
+            // parse failure on coinbaseSize, must be address
+            destStr = params[0].get_str();
```

Comment:
This is not how optional positional parameters work 🤔

Reply:
Indeed. Note that I modified the PR now to accept:

`getminincandidate` # no coinbaseSize specified, use default address (daemon wallet)
`getminingcandidate 1024` # specify custom coinbaseSize
`getminingcandidate 1024 bchtest:bla...` # use custom coinbaseSize, custom address
`getminingcandidate null bchtest:bla...` # use **default** coinbaseSize, custom address
^ This is new

I hope this is more in line with how things are expected to work. Note the only change here to the RPC API would be before `null` was rejected as a coinbaseSize (it had no meaning) -- and now it means "use default from daemon tweaks".